athletics.lic

```diff
@@ -0,0 +1,170 @@
+    end
+  end
+  
+  def done_training?
+    return DRSkill.getxp('Athletics') >= @end_skill
+  end
+  
+  def crossing_athletics
+    if DRSkill.getrank('Athletics') >= 400
+      climb_branch
+    else
+      until done_training?
+        @athletics_options
+          .reject { |data| @settings.avoid_athletics_in_justice && data['justice'] }
+          .each do |data|
+            break unless climb?(data['room'], data['targets'])
+        end
+      end
+    end
+  end
+  
+  def shard_athletics
+    until done_training?
+      if DRSkill.getrank('Athletics') >120 and DRSkill.getrank('Athletics') <= 250
+        swim_zaldeni   
+      elsif DRSkill.getrank('Athletics') < 550
+        if DRSkill.getrank('Stealth') >= 250
+          #Shard side of undergondola.  Need to hide to practice there
+          climb_practice(19460,"wall",true)
+        else
+          swim_liirewsag
+        end
+      else
+        climb_branch
+      end
+    end
+  end
+  
+  def hib_athletics
+    swim_liirewsag
+  end
+  
+  def ratha_athletics
+    until done_training?
+      if DRSkill.getrank('Athletics') <= 185
+        climb_practice(5287, 'rock gorge',true)
+      else
+        if DRSkill.getrank('Stealth') >= 130
+          climb_practice(5143, 'deep crack',true)
+        end
+      end
+    end
+  end
+  
+  def swim_zaldeni
+    until done_training?
+      walk_to(11481)
+      walk_to(11482)
+      walk_to(11483)
+    end 
+  end
+  
+  def swim_liirewsag
+    #Intermittent errors in the map will cause go2 to get stuck.  The manual moves are a workaround for that
+    walk_to(4155)
+    move('nw')
+    until done_training?
+      fput('climb bank')
+      3.times do
+        fput('south')
+        waitrt?
+      end 
+      walk_to(4155)
+      move('nw')
+    end
+    fput('climb bank')
+    waitrt?
+  end
+
+  def climb_practice(room, target, to_hide = false)
+    return unless target
+    walk_to(room)
+    until done_training?
+      retreat
+      if to_hide
+        remaining_attempts = 5
+        while remaining_attempts > 0
+          break if hide?
+          remaining_attempts -= 1
+        end
+        return if remaining_attempts.zero?
+      end
+      Flags.add('ct-climbing-finished', 'You finish practicing your climbing')
+      Flags.add('ct-climbing-combat', 'You are engaged in combat')
+      bput("climb practice #{target}", 'You begin to practice ')
+      loop do
+        pause
+        break if Flags['ct-climbing-finished']
+        return if Flags['ct-climbing-combat']
+        break if done_training?
+      end
+      DRC.bput('stop climb','You stop practicing your climbing skills.',"You weren't practicing your climbing skills anyway.")
+    end
+    
+    bput('unhide', 'You come out of hiding', 'You slip out of hiding', 'But you are not') if to_hide
+  end
+  
+  def climb?(room, targets)
+    targets.each do |target|
+      walk_to(room)
+      return true if DRRoom.npcs.length >= 3
+
+      fput "climb #{target}"
```

Comment:
There isn't any messaging on a successful climb, it just moves you to the next room, so the only bput I could use matches a wildcard